internal/circle/service.go

```diff
@@ -103,7 +111,7 @@ func (s *Service) getClient(ctx context.Context) (ClientInterface, error) {
 	if err != nil {
 		return nil, fmt.Errorf("retrieving decrypted Circle API key: %w", err)
 	}
-	return s.ClientFactory(s.NetworkType, apiKey), nil
+	return s.ClientFactory(s.NetworkType, apiKey, s.TenantManager), nil
```

Comment:
🤔  Now that I'm reading this class again, I gotta say it's not 100% clear that the client being instantiated here uses the apiKey from the tenant that is found in the context.

I know this is not something you introduced, but could you improve the naming of the function to make it clearer?Maybe `getClientForTenantInContext` or `getClientForContextTenant`, or something like that?